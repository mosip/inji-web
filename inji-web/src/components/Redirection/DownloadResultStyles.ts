export const DownloadResultStyles = {
    container: "bg-white rounded-lg shadow-iw flex flex-col flex-grow h-fit items-center justify-center"
}