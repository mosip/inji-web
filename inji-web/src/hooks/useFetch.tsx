import {useState} from "react";
import {api, MethodType} from "../utils/api";

export enum RequestStatus {
    LOADING,
    DONE,
    ERROR
}

export const useFetch = () => {
    const [state, setState] = useState<RequestStatus>(RequestStatus.LOADING);
    const [error, setError] = useState<string>("");

    const fetchRequest = async (uri: string, method: MethodType, header: any, body?: any) => {
        try {
            setState(RequestStatus.LOADING);
            const response = await fetch(`${uri}`, {
                method: MethodType[method],
                headers: header,
                body: body,
            });

            if (uri.indexOf("download") !== -1) {
                setState(RequestStatus.DONE);
                return await response.blob();
            }
            if (!response.ok) {
                setState(RequestStatus.ERROR);
                return response;
            }
            setState(RequestStatus.DONE);
            return await response.json();
        } catch (e) {
            setState(RequestStatus.ERROR);
            setError("Error Happened");
        }
    };
    return {state, error, fetchRequest};
}



