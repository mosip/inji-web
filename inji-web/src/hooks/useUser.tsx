import {useState, useEffect} from "react";
import { api } from "../utils/api";

export type User = {
    displayName: string;
    profilePictureUrl: string;
}

type ErrorObj = {
    errorCode: string;
    errorMessage: string;
}

export function useUser() {
    const [user, setUser] = useState<User | null>(null);
    const [errorObj, setErrorObj] = useState<ErrorObj | null>(null);

    useEffect(() => {
        const userData = localStorage.getItem("user");
        if (userData) {
            try {
                setUser(JSON.parse(userData));
            } catch (e) {
                console.error("Invalid user data in localStorage");
                localStorage.removeItem("user");
            }
        }
    }, []);

    const saveUser = (userData: User) => {
        localStorage.setItem("user", JSON.stringify(userData));
        setUser(userData);
    };

    const removeUser = () => {
        localStorage.removeItem("user");
        setUser(null);
    };

    const fetchUserProfile = async () => {
        try {
            const response = await fetch(api.fetchUserProfile.url(), {
                method: api.fetchUserProfile.methodType === 0 ? "GET" : "POST",
                headers: {...api.fetchUserProfile.headers()},
                credentials: "include"
            });
    
            const responseData = await response.json();
    
            if (!response.ok) {
                throw responseData;
            }
    
            const userData: User = {
                "displayName": responseData.display_name,
                "profilePictureUrl": responseData.profile_picture_url
            };

            saveUser(userData);
        } catch (error) {
            console.error("Error occurred while fetching user profile:", error);
            setUser(null);
            setErrorObj(error as ErrorObj)
            throw error;
        }
    };

    return {user, errorObj, saveUser, removeUser, fetchUserProfile};
}
