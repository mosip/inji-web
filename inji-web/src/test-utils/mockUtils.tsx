import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, BrowserRouter as Router, Routes } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

// Mock for storage module
export const mockStorageModule = () => {
    jest.mock('../utils/storage.ts', () => ({
        storage: {
            getItem: jest.fn(),
            setItem: jest.fn(),
            SESSION_INFO: 'SESSION_INFO',
            SELECTED_LANGUAGE: 'selectedLanguage'
        }
    }));
};

// Mock for localStorage
export const mockLocalStorage = () => {
    let store: { [key: string]: string } = {};
  
    const localStorageMock = {
      getItem: jest.fn((key: string) => store[key] || null),
      setItem: jest.fn((key: string, value: string) => {
        store[key] = value;
      })
    };
  
    Object.defineProperty(window, 'localStorage', { 
      value: localStorageMock,
      writable: true
    });
  
    return localStorageMock;
  };

// API Mock
export const mockApi = {
    authorizationRedirectionUrl: 'http://mockurl.com'
};

// Crypto Mock
export const mockCrypto = {
    getRandomValues: (array: Uint32Array) => {
        for (let i = 0; i < array.length; i++) {
            array[i] = 4294967296;
        }
        return array;
    },
    subtle: {} as SubtleCrypto,
    randomUUID: () => '123e4567-e89b-12d3-a456-426614174000'
} as Crypto;

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    const mockNavigate = jest.fn();
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: mockNavigate,
    }));
};

export const mockUseSelector = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useSelector: jest.fn(),
    }));
};

export const setMockUseSelectorState = (state: any) => {
    const useSelectorMock = require('react-redux').useSelector;
    useSelectorMock.mockImplementation((selector: any) => selector(state));
};

export const mockUseGetObjectForCurrentLanguage = () => {
    jest.mock('../utils/i18n', () => ({
        getObjectForCurrentLanguage: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

export const mockUseSearchCredentials = () => {
    jest.mock('../components/Credentials/SearchCredential', () => ({
        SearchCredential: () => <></>,
    }));
 };

export const mockUseDispatch = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useDispatch: jest.fn(),
    }));
};
              
export const mockUseSpinningLoader = () => {
    jest.mock('../components/Common/SpinningLoader', () => ({
        SpinningLoader: () => <></>,
    }));
};

export const mockUseLanguageSelector = () => {
    jest.mock('../components/Common/LanguageSelector', () => ({
        LanguageSelector: () => <></>,
    }));
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

export const renderWithProvider = (element: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {element}
            </Router>
        </Provider>,
        options
    );
};
    
export const mockUseParams = ()=>{
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useParams: () => ({ issuerId: 'test-issuer-id' }),
      }));   
};

export const mockUseapiObject = () =>{
    jest.mock('../utils/api', () => ({
        api: {
          mimotoHost: 'http://mocked-api-host',
        },
      }));
};

export const mockUseFetch = () =>{
    jest.mock('../hooks/useFetch', () => {
        const RequestStatus = {
          LOADING: 'LOADING',
          DONE: 'DONE',
          ERROR: 'ERROR',
        };
        return {
          useFetch: () => jest.fn(),
          RequestStatus,
        };
      });
};

export const mockUseToast = () =>{
    jest.mock('react-toastify', () => ({
      toast: {
        error: jest.fn(),
      },
    }));
};

export const mockusemisc = ()=>{
    jest.mock('../utils/misc', () => ({
        downloadCredentialPDF: jest.fn(),
        getErrorObject: jest.fn(),
        getTokenRequestBody: jest.fn(),
      }));
};

export const mockusei18n = ()=>{
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
          t: (key: string) => key,
        }),
        initReactI18next: {
          type: '3rdParty',
          init: jest.fn(),
        },
      }));

};

export const mockWindowLocation = (url: string) => {
    const location = new URL(url) as unknown as Location;
    Object.defineProperty(window, 'location', {
      value: location,
      writable: true,
    });
  };

export const renderWithRouter = (Element: React.ReactElement, { route = '/' } = {}) => {
    window.history.pushState({}, 'Test page', route);
    return render(
      <BrowserRouter>
        <Provider store={reduxStore}>
          <Routes>
            <Route path="*" element={Element} />
          </Routes>
        </Provider>
      </BrowserRouter>
    );
  };

