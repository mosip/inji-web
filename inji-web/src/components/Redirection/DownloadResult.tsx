import React from "react";
import {RequestStatus} from "../../hooks/useFetch";
import {SpinningLoader} from "../Common/SpinningLoader";
import {ErrorSheildIcon} from "../Common/ErrorSheildIcon";
import {SuccessSheildIcon} from "../Common/SuccessSheildIcon";
import {LandingPageWrapper} from "../Common/LandingPageWrapper";
import {useUser} from "../../hooks/useUser";
import {DownloadResultStyles} from "./DownloadResultStyles";

interface StateConfig {
    icon: JSX.Element;
    gotoHome: boolean;
}

interface DownloadResultProps {
    state: RequestStatus;
    title: string;
    subTitle?: string;
}

export const DownloadResult: React.FC<DownloadResultProps> = ({title, subTitle, state}) => {
    const {isUserLoggedIn} = useUser();

    const stateConfig: Record<RequestStatus, StateConfig> = {
        [RequestStatus.DONE]: {
            icon: <SuccessSheildIcon/>,
            gotoHome: true,
        },
        [RequestStatus.ERROR]: {
            icon: <ErrorSheildIcon/>,
            gotoHome: true,
        },
        [RequestStatus.LOADING]: {
            icon: <SpinningLoader/>,
            gotoHome: false,
        },
    };

    const currentConfig = stateConfig[state];

    const baseWrapperProps = {
        icon: currentConfig.icon,
        title: title,
        subTitle: subTitle,
        gotoHome: currentConfig.gotoHome,
    };


    return (
        isUserLoggedIn ? (
            <div
                className={DownloadResultStyles.container}>
                <LandingPageWrapper{...baseWrapperProps}/>
            </div>
        ) : (
            <LandingPageWrapper{...baseWrapperProps}/>
        )
    );
};