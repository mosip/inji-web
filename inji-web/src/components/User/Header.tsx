import React, {useEffect, useRef, useState} from 'react';
import {useNavigate} from 'react-router-dom';
import {LanguageSelector} from '../Common/LanguageSelector';
import {api} from '../../utils/api';
import {toast} from 'react-toastify';
import {useUser} from '../../hooks/User/useUser';
import HamburgerMenu from '../../assets/HamburgerMenu.svg';
import {isRTL} from '../../utils/i18n';
import {useSelector} from 'react-redux';
import {RootState} from '../../types/redux';
import {useTranslation} from 'react-i18next';
import DropdownArrowIcon from '../Common/DropdownArrowIcon';
import {ROUTES} from '../../utils/constants';
import {convertStringIntoPascalCase} from "../../utils/misc";
import {navigateToUserHome} from "../../utils/navigationUtils";
import {CircleSkeleton} from '../Common/CircleSkeleton';
import {InfoFieldSkeleton} from '../Common/InfoFieldSkeleton';
import {ApiError, DropdownItem, ResponseTypeObject} from "../../types/data";
import {useApi} from "../../hooks/useApi";

type UserHeaderProps = {
    headerRef: React.RefObject<HTMLDivElement>;
    headerHeight: number;
};

export const Header: React.FC<UserHeaderProps> = ({
    headerRef,
    headerHeight
}) => {
    const language = useSelector((state: RootState) => state.common.language);
    const navigate = useNavigate();
    const [displayName, setDisplayName] = useState<string | undefined>(
        undefined
    );
    const [isProfileDropdownOpen, setIsProfileDropdownOpen] = useState(false);
    const [isHamburgerMenuOpen, setIsHamburgerMenuOpen] = useState(false);
    const dropdownRef = useRef<HTMLDivElement>(null);
    const hamburgerMenuRef = useRef<HTMLImageElement>(null);
    const {user, removeUser,isLoading} = useUser();
    const displayNameFromLocalStorage = user?.displayName;
    const hasProfilePictureUrl = user?.profilePictureUrl;
    const {t} = useTranslation('User');
    const logoutRequest = useApi()

    useEffect(() => {
        setDisplayName(displayNameFromLocalStorage);
    }, [displayNameFromLocalStorage]);

    useEffect(() => {
        const handleClickOutside = (event: MouseEvent) => {
            const target = event.target as Node;

            if (dropdownRef.current && !dropdownRef.current.contains(target)) {
                setIsProfileDropdownOpen(false);
            }

            if (
                hamburgerMenuRef.current &&
                !hamburgerMenuRef.current.contains(target)
            ) {
                setIsHamburgerMenuOpen(false);
            }
        };

        document.addEventListener('click', handleClickOutside);
        return () =>
            document.removeEventListener('click', handleClickOutside);
    }, []);

    const handleLogout = async () => {
        try {
            const response = await logoutRequest.fetchData({
                apiRequest: api.userLogout,
            })

            if (response.ok()) {
                removeUser();
                window.location.replace(ROUTES.ROOT);
            } else {
                const parsedResponse = ((response.error as ApiError)?.response?.data as ResponseTypeObject).errors
                const errorCode = parsedResponse?.[0].errorCode;
                if (errorCode === 'user_logout_error') {
                    removeUser();
                    window.location.replace(ROUTES.ROOT);
                }
                throw new Error(parsedResponse?.[0]?.errorMessage);
            }
        } catch (error) {
            console.error('Logout failed with error:', error);
            toast.error('Unable to logout. Please try again.');
        }
    };

    const dropdownItems: DropdownItem[] = [
        {
            label: t('ProfileDropdown.profile'),
            onClick: () => {
                setIsProfileDropdownOpen(false);    
                navigate(ROUTES.PROFILE,{state: {from: window.location.pathname}});
            },
            textColor: 'text-gray-700',
            key: 'Profile-Dropdown-Profile'
        },
        {
            label: t('ProfileDropdown.faq'),
            onClick: () => {
                setIsProfileDropdownOpen(false);
                navigate(ROUTES.USER_FAQ, {state: {from: window.location.pathname}});
            },
            textColor: 'text-gray-700',
            key: 'Profile-Dropdown-FAQ'
        },
        {
            label: t('ProfileDropdown.logout'),
            onClick: handleLogout,
            textColor: 'text-red-700',
            key: 'Profile-Dropdown-Logout'
        }
    ];

    const toggleProfileDropdown = (e: React.MouseEvent<HTMLDivElement>) => {
        e.stopPropagation(); // Prevent global click handler from immediately closing the dropdown
        setIsProfileDropdownOpen(!isProfileDropdownOpen);
    };

    const toggleHamburgerMenu = () => {
        setIsHamburgerMenuOpen(!isHamburgerMenuOpen);
    };

    const getProfileInitials = (displayName: string | undefined) => {
        return displayName ? displayName.charAt(0).toUpperCase(): 'U';
    };

    const getUserProfileIconWithName = () => {
        if (isLoading) {
            return (
              <div className="flex gap-2 items-center">
                <CircleSkeleton size="w-12 h-12" />
                <InfoFieldSkeleton width="w-24" height="h-2" />
              </div>
            );
          }
        
          return (
            <div className="flex gap-2 items-center">
              <div
                className={`aspect-square w-12 rounded-full bg-iw-avatarPlaceholder overflow-hidden flex items-center justify-center text-iw-avatarText font-medium text-lg ${
                  !hasProfilePictureUrl && 'p-2 sm:p-3 md:p-4'
                }`}
              >
                {hasProfilePictureUrl ? (
                  <img
                    src={user.profilePictureUrl}
                    alt="Profile Pic"
                    className="w-12 h-12 object-cover rounded-full"
                    referrerPolicy="no-referrer"
                  />
                ) : (
                  getProfileInitials(displayName)
                )}
              </div>
              <span className="font-semibold text-gray-800">
                {convertStringIntoPascalCase(displayName)}
              </span>
            </div>
          );
        };

    return (
        <header
            ref={headerRef}
            data-testid="dashboard-header-container"
            className="fixed top-0 left-0 right-0 z-20 bg-iw-background bg-transparent shadow-[0_4px_5px_0_rgba(0,0,0,0.051)]"
        >
            <div className="container mx-auto px-4 py-4 flex justify-between items-center">
                <div className="flex justify-start items-center gap-2">
                    <div
                        data-testid="hamburger-menu"
                        className="block sm:hidden"
                    >
                        <img
                            data-testid="icon-hamburger-menu"
                            ref={hamburgerMenuRef}
                            src={HamburgerMenu}
                            alt="Hamburger Menu"
                            onClick={toggleHamburgerMenu}
                            className="cursor-pointer"
                        />
                    </div>
                    <div
                        className="w-[130px] sm:w-[160px] md:w-[170px]"
                        role={'button'}
                        tabIndex={0}
                        onMouseDown={() => navigateToUserHome(navigate)}
                        onKeyUp={() => navigateToUserHome(navigate)}
                    >
                        <img
                            src={require('../../assets/InjiWebLogo.png')}
                            className={`max-w-full h-auto object-contain cursor-pointer ${
                                isRTL(language) ? 'mr-4' : ''
                            }`}
                            data-testid="header-injiWeb-logo"
                            alt="Inji Web Logo"
                        />
                    </div>
                </div>

                <div className="flex items-center space-x-6">
                    <LanguageSelector />

                    <div
                        data-testid="profile-details"
                        className="hidden sm:block relative"
                        ref={dropdownRef}
                    >
                        <div className="flex items-center space-x-2">
                            {getUserProfileIconWithName()}
                            {!isLoading && (
                                <div
                                className="relative inline-block cursor-pointer"
                                onClick={toggleProfileDropdown}
                                >
                                <DropdownArrowIcon isOpen={isProfileDropdownOpen} />
                                </div>
                            )}
                        </div>

                        {isProfileDropdownOpen && (
                            <div
                                data-testid="profile-dropdown"
                                className="absolute -right-7 top-100 mt-8 w-56 bg-white rounded-lg shadow-lg z-50 font-medium"
                            >
                                <div className="absolute top-[-0.3rem] right-8 w-3 h-3 bg-white transform rotate-45" />
                                <div className="py-2">
                                    {dropdownItems.map((item, index) => (
                                        <React.Fragment key={item.key}>
                                            <div
                                                className={`px-4 py-2 text-sm ${item.textColor} hover:bg-gray-100 cursor-pointer`}
                                                onClick={item.onClick}
                                            >
                                                {item.label}
                                            </div>
                                            {index !==
                                                dropdownItems.length - 1 && (
                                                <hr className="border-gray-200 my-1 mx-2" />
                                            )}
                                        </React.Fragment>
                                    ))}
                                </div>
                            </div>
                        )}
                    </div>
                </div>
            </div>
            <div
                data-testid="hamburger-menu-dropdown"
                className="block sm:hidden w-full"
            >
                {isHamburgerMenuOpen && (
                    <div
                        style={{marginTop: headerHeight}}
                        className="absolute top-1 bg-white shadow-iw-hamburger-dropdown p-2 w-full"
                    >
                        <div>
                            <div className="flex items-center px-4 py-2">
                                {getUserProfileIconWithName()}
                            </div>
                            {dropdownItems.map((item, index) => (
                                <React.Fragment key={item.key}>
                                    <div
                                        className={`px-4 py-2 text-sm ${item.textColor} hover:bg-gray-100 cursor-pointer font-medium`}
                                        onClick={item.onClick}
                                    >
                                        {item.label}
                                    </div>
                                    {index !== dropdownItems.length - 1 && (
                                        <hr className="border-gray-200 my-1 mx-2" />
                                    )}
                                </React.Fragment>
                            ))}
                        </div>
                    </div>
                )}
            </div>
        </header>
    );
};
