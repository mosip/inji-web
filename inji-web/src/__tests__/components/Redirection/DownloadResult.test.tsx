import {renderWithProvider} from '../../../test-utils/mockUtils';
import {DownloadResult} from '../../../components/Redirection/DownloadResult';
import {RequestStatus} from '../../../hooks/useFetch';
import {screen} from "@testing-library/react";
import {useUser} from "../../../hooks/User/useUser";

const mockLandingPageWrapper = jest.fn();
jest.mock('../../../components/Common/LandingPageWrapper', () => ({
    LandingPageWrapper: (props: any) => {
        mockLandingPageWrapper(props);
        return <div data-testid="mock-landing-page-wrapper" {...props} />;
    },
}));

jest.mock("../../../hooks/User/useUser.tsx", () => {
    const actualModule = jest.requireActual('../../../hooks/User/useUser');
    return {
        useUser: jest.fn(),
        UserProvider: actualModule.UserProvider,
    };
});

const mockUseUser = useUser as jest.Mock;

describe('Testing of DownloadResult -> ', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        mockUseUser.mockReturnValue({isUserLoggedIn: () => false});
    });

    it('matches snapshot when user is not logged in', () => {
        const {asFragment} = renderWithProvider(
            <DownloadResult title="Title" subTitle="SubTitle" state={RequestStatus.DONE}/>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot when user is logged in', () => {
        mockUseUser.mockReturnValue({isUserLoggedIn: () => true});

        const {asFragment} = renderWithProvider(
            <DownloadResult title="Title" subTitle="SubTitle" state={RequestStatus.DONE}/>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders div wrapper when user is logged in', () => {
        mockUseUser.mockReturnValue({isUserLoggedIn: () => true});

        renderWithProvider(<DownloadResult title="Title" subTitle="SubTitle" state={RequestStatus.DONE}/>);

        expect(screen.getByTestId('download-result-container')).toBeInTheDocument();
        expect(screen.getByTestId('mock-landing-page-wrapper')).toBeInTheDocument();
    });

    it('does not render div wrapper when user is not logged in', () => {
        renderWithProvider(
            <DownloadResult title="Title" subTitle="SubTitle" state={RequestStatus.DONE}/>
        );

        expect(screen.queryByTestId('download-result-container')).not.toBeInTheDocument();
        expect(screen.getByTestId('mock-landing-page-wrapper')).toBeInTheDocument();
    });

    test.each([
        [
            'DONE',
            RequestStatus.DONE,
            {icon: expect.anything(), title: 'Title', subTitle: 'SubTitle', gotoHome: true},
        ],
        [
            'ERROR',
            RequestStatus.ERROR,
            {icon: expect.anything(), title: 'Title', subTitle: 'SubTitle', gotoHome: true},
        ],
        [
            'LOADING',
            RequestStatus.LOADING,
            {icon: expect.anything(), title: 'Title', subTitle: 'SubTitle', gotoHome: false},
        ],
    ])(
        'should call LandingPageWrapper with correct props for state %s',
        (_stateName, state, expectedProps) => {
            renderWithProvider(<DownloadResult title="Title" subTitle="SubTitle" state={state}/>);

            expect(mockLandingPageWrapper).toHaveBeenCalledWith(
                expect.objectContaining(expectedProps)
            );
        }
    );
});