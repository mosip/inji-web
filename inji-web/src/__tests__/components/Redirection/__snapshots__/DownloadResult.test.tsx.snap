// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout of DownloadResult for Success Error and Loading Check if the layout is matching with the snapshots for the Error 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center pt-32"
    data-testid="DownloadResult-Outer-Container"
  >
    <div
      class="rounded-full p-2 shadow bg-white"
    >
      <div
        class="rounded-full p-8 bg-iw-shieldErrorShadow"
      >
        <svg
          color="var(--iw-color-shieldErrorIcon)"
          data-testid="DownloadResult-Error-ShieldIcon"
          fill="currentColor"
          height="40"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0c-.69 0-1.843.265-2.928.56-1.11.3-2.229.655-2.887.87a1.54 1.54 0 0 0-1.044 1.262c-.596 4.477.787 7.795 2.465 9.99a11.8 11.8 0 0 0 2.517 2.453c.386.273.744.482 1.048.625.28.132.581.24.829.24s.548-.108.829-.24a7 7 0 0 0 1.048-.625 11.8 11.8 0 0 0 2.517-2.453c1.678-2.195 3.061-5.513 2.465-9.99a1.54 1.54 0 0 0-1.044-1.263 63 63 0 0 0-2.887-.87C9.843.266 8.69 0 8 0M6.854 5.146 8 6.293l1.146-1.147a.5.5 0 1 1 .708.708L8.707 7l1.147 1.146a.5.5 0 0 1-.708.708L8 7.707 6.854 8.854a.5.5 0 1 1-.708-.708L7.293 7 6.146 5.854a.5.5 0 1 1 .708-.708"
          />
        </svg>
      </div>
    </div>
    <div
      class="my-4"
      data-testid="DownloadResult-Title-Container"
    >
      <p
        class="font-bold"
        data-testid="DownloadResult-Title"
      >
        Title
      </p>
    </div>
    <div
      class="mb-6 px-10 text-center"
      data-testid="DownloadResult-SubTitle-Container"
    >
      <p
        data-testid="DownloadResult-SubTitle"
      >
        SubTitle
      </p>
    </div>
    <div
      class=""
    >
      <div
        class="bg-gradient-to-r rounded-lg text-center cursor-pointer shadow hover:shadow-lg from-iw-primary to-iw-secondary p-0.5"
      >
        <div
          class="py-2 px-4 rounded-lg bg-white"
        >
          <button
            class="text-iw-tertiary font-bold rounded-lg"
            data-testid="DownloadResult-Home-Button"
          >
            <span
              class="bg-gradient-to-r from-iw-primary to-iw-secondary bg-clip-text text-md font-bold text-transparent"
            >
              Go To Home
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Layout of DownloadResult for Success Error and Loading Check if the layout is matching with the snapshots for the Loading 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center pt-32"
    data-testid="DownloadResult-Outer-Container"
  >
    <div
      class="flex justify-center items-center"
      data-testid="SpinningLoader-Container"
    >
      <div
        aria-busy="true"
        aria-label="oval-loading"
        class="sc-beySPh gZmPOF"
        data-testid="oval-loading"
        role="progressbar"
      >
        <svg
          data-testid="oval-svg"
          height="80"
          stroke="var(--iw-color-spinningLoaderPrimary)"
          viewBox="-20.5 -20.5 43 43"
          width="80"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <g
              data-testid="oval-secondary-group"
              stroke-width="3"
              transform="translate(1 1)"
            >
              <circle
                cx="0"
                cy="0"
                r="20"
                stroke="var(--iw-color-spinningLoaderSecondary)"
                stroke-opacity=".5"
                stroke-width="3"
              />
              <path
                d="M20 0c0-9.94-8.06-20-20-20"
              >
                <animatetransform
                  attributeName="transform"
                  dur="1s"
                  from="0 0 0"
                  repeatCount="indefinite"
                  to="360 0 0"
                  type="rotate"
                />
              </path>
            </g>
          </g>
        </svg>
      </div>
    </div>
    <div
      class="my-4"
      data-testid="DownloadResult-Title-Container"
    >
      <p
        class="font-bold"
        data-testid="DownloadResult-Title"
      >
        Title
      </p>
    </div>
    <div
      class="mb-6 px-10 text-center"
      data-testid="DownloadResult-SubTitle-Container"
    >
      <p
        data-testid="DownloadResult-SubTitle"
      >
        SubTitle
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Layout of DownloadResult for Success Error and Loading Check if the layout is matching with the snapshots for the Success 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center pt-32"
    data-testid="DownloadResult-Outer-Container"
  >
    <div
      class="rounded-full p-2 shadow bg-white"
    >
      <div
        class="rounded-full p-8 bg-iw-shieldSuccessShadow "
      >
        <svg
          color="var(--iw-color-shieldSuccessIcon)"
          data-testid="DownloadResult-Success-ShieldIcon"
          fill="currentColor"
          height="40"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0c-.69 0-1.843.265-2.928.56-1.11.3-2.229.655-2.887.87a1.54 1.54 0 0 0-1.044 1.262c-.596 4.477.787 7.795 2.465 9.99a11.8 11.8 0 0 0 2.517 2.453c.386.273.744.482 1.048.625.28.132.581.24.829.24s.548-.108.829-.24a7 7 0 0 0 1.048-.625 11.8 11.8 0 0 0 2.517-2.453c1.678-2.195 3.061-5.513 2.465-9.99a1.54 1.54 0 0 0-1.044-1.263 63 63 0 0 0-2.887-.87C9.843.266 8.69 0 8 0m2.146 5.146a.5.5 0 0 1 .708.708l-3 3a.5.5 0 0 1-.708 0l-1.5-1.5a.5.5 0 1 1 .708-.708L7.5 7.793z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      class="my-4"
      data-testid="DownloadResult-Title-Container"
    >
      <p
        class="font-bold"
        data-testid="DownloadResult-Title"
      >
        Title
      </p>
    </div>
    <div
      class="mb-6 px-10 text-center"
      data-testid="DownloadResult-SubTitle-Container"
    >
      <p
        data-testid="DownloadResult-SubTitle"
      >
        SubTitle
      </p>
    </div>
    <div
      class=""
    >
      <div
        class="bg-gradient-to-r rounded-lg text-center cursor-pointer shadow hover:shadow-lg from-iw-primary to-iw-secondary p-0.5"
      >
        <div
          class="py-2 px-4 rounded-lg bg-white"
        >
          <button
            class="text-iw-tertiary font-bold rounded-lg"
            data-testid="DownloadResult-Home-Button"
          >
            <span
              class="bg-gradient-to-r from-iw-primary to-iw-secondary bg-clip-text text-md font-bold text-transparent"
            >
              Go To Home
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
